### DISCUSSION TITLE
Build Error: import "unix-bytestring" System.Posix.IO.ByteString (fdPreadBuf)

### DISCUSSION THREAD

**Discussion by jritchie-nullable** [IntersectMBO/cardano-node#5019] [General]
Ran into this build error and  reported it here: https://github.com/input-output-hk/ouroboros-network/issues/4471

Adding this here for visibility.

**Comment by newhoggy** [comment]
Perhaps there are missing steps in the documentation for building `cardano-node`?  What OS are you using and which documentation did you use?

**Reply by jritchie-nullable** [reply]
@newhoggy Hi, it looks like the unix-bytestring library has the breaking change.  ouroboros-consensus did not have the version of the library pinned but they have since removed that dependency in their master.  I think any clean builds (maybe delete your ~/.cabal) will fail. I'm going to test this on linux, the failure first appeared on a clean MacOS build for me.
